src/smc-webapp/markdown.ts

```diff
@@ -48,14 +58,36 @@ export function markdown_to_html(markdown_string: string): string {
   // console.log(2, JSON.stringify(text), JSON.stringify(math));
   // Process checkboxes [ ].
   text = checkboxes(text);
-  // Render text to HTML.
-  let html: string = markdown_it.render(text);
+
+  let html: string;
+  let frontmatter = "";
+
+  // avoid instantiating a new markdown object for normal md processing
+  if (mode == "frontmatter") {
+    const md_frontmatter = new MarkdownIt(OPTIONS).use(
+      MarkdownItFrontMatter,
+      fm => {
+        frontmatter = fm;
```

Comment:
this is weird code, but it's synchronous and that frontmatter processor still has a function as an argument. using a module-level variable `frontmatter` is likely not a safe thing to do...